massa-execution-worker/src/speculative_ledger.rs

```diff
@@ -206,6 +314,42 @@ impl SpeculativeLedger {
             )));
         }
```

Comment:
note: set_bytecode should probably also get the "caller_addr" address to charge for the costs if it's called from an addr A to change the bytecode of an addr B (happens on SC initialization for example)

Reply:
There is a special function for this case called create_new_sc_address in speculative ledger.

Reply:
there is no set_bytecode_for ABI ?

Reply:
Yes but the address pass as parameter should pay the cost for his proper storage costs

Reply:
not if it is being initialized by someone else (see https://github.com/massalabs/massa/pull/3043#discussion_r981393381 )